src/net/tcp/stream.rs

```diff
@@ -93,6 +93,21 @@ impl TcpStream {
         Ok(TcpStream::new(pair, rx))
     }
 
+    /// Try to write a buffer to the stream, returning how many bytes were
+    /// written.
+    ///
+    /// The function will attempt to write the entire contents of `buf`, but
+    /// only part of the buffer may be written.
+    ///
+    /// # Return
+    ///
+    /// If data is successfully written, `Ok(n)` is returned, where `n` is the
+    /// number of bytes written. If the stream has been shut down
```

Comment:
nit: I'd just copy the doc verbatim. In future changes this will do what the tokio flavor does and then we don't need to update the doc.

Reply:
Missed this nit. You're probably right, although I see some (small) benefit in the doc representing the current state